pkg/lib/cockpit-components-table.jsx

```diff
@@ -228,11 +229,12 @@ export const ListingTable = ({
                             }
                         }} />
                     }
-                    {row.columns.map((cell, cellIndex) => {
+                    {row.columns.map(cell => {
                         const { key, ...cellProps } = cell.props || {};
-                        const dataLabel = typeof cells[cellIndex] == 'object' ? cells[cellIndex].title : cells[cellIndex];
-                        const colKey = dataLabel || cellIndex;
-                        if (cells[cellIndex]?.header)
+                        const dataLabel = typeof cells[columnSpanCnt] == 'object' ? cells[columnSpanCnt].title : cells[columnSpanCnt];
+                        columnSpanCnt += cellProps.colSpan || 1;
```

Comment:
I find a map with side-effects to be a bit distasteful, but I'm not sure if it's common practice in JS or not...

Reply:
I went shopping in lodash for a primitive that would allow for something like this:

```ts
    const arr = [1, 2, 3];
    stateful_map(arr, item, state => [-(item + state), item], 0);
    > [-1, -3, -5]
```

ie: where each iteration allows returning a value but also allows forwarding a state value to the next iteration.

I couldn't find it, and I doubt you'd want to roll your own.

I defer to your judgment here.

Reply:
I would not like to complicate the logic. I dont think the current implementation is an antipractice and it's rather simple.